logstash-core/lib/logstash/api/commands/stats.rb

```diff
@@ -183,7 +184,12 @@ def report(stats, extended_stats = nil, opts = {})
                 :outputs => plugin_stats(stats, :outputs)
               },
               :reloads => stats[:reloads],
-              :queue => stats[:queue]
+              :queue => stats[:queue],
+              :pipeline => {
+                :workers => pipeline_config[:workers],
+                :batch_size => pipeline_config[:batch_size],
+                :batch_delay => pipeline_config[:batch_delay],
```

Comment:
```suggestion
                :workers => stats[:config][:workers],
                :batch_size => stats[:config]:batch_size],
                :batch_delay => stats[:config][:batch_delay],
```
I'm not following why we add a new parameter and do a specific lookup in the caller when we appear to have this data already in the `stats` object. Is there something special about `get_shallow` that provides defaults or something?

Reply:
I guess my eyes were a bit blur when I read a chain of `pipelines` and `pipeline` and thought `stats` is from somewhere else. Good catch!